integration/container/create_test.go

```diff
@@ -91,3 +92,51 @@ func TestCreateWithInvalidEnv(t *testing.T) {
 		})
 	}
 }
+
+// Test case for #30166 (target was not validated)
+func TestCreateTmpfsMountsTarget(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType != "linux")
+
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+
+	testCases := []struct {
+		target        string
+		expectedError string
+	}{
+		{
+			target:        ".",
+			expectedError: "mount path must be absolute",
+		},
+		{
+			target:        "foo",
+			expectedError: "mount path must be absolute",
+		},
+		{
+			target:        "/",
+			expectedError: "destination can't be '/'",
+		},
+		{
+			target:        "//",
+			expectedError: "destination can't be '/'",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.target, func(t *testing.T) {
+			t.Parallel()
```

Comment:
Can't use parallel with the deferred cleanup.

Reply:
@cpuguy83 Update. Thanks.